src/resolve/resolved-dns-packet.c

```diff
@@ -2174,6 +2174,8 @@ int dns_packet_extract(DnsPacket *p) {
                                 if (r < 0)
                                         goto finish;
                         }
+
+                        previous = rr;
```

Comment:
It seems strange to me (a casual observer's perspective) that previous gets updated here.  There's a bunch of short-circuiting continues which will cause reentry of dns_resource_key_reduce() above with the same previous from the last iteration.  Just make sure this is as intended...